src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -103,7 +103,11 @@ public static unsafe string GetTempFileName()
             int tempPathByteCount = Encoding.UTF8.GetByteCount(tempPath);
             int totalByteCount = tempPathByteCount + fileTemplate.Length + 1;
 
+#if TARGET_BROWSER
+            Span<byte> path = new byte[totalByteCount];
```

Comment:
Mention the Emscripten issue here in a comment.
Maybe also add that the workaround for that is to allocate the buffer on the heap, so we always have a different pointer. Please correct me if I misunderstood.